signbank/dictionary/adminviews.py

```diff
@@ -231,9 +231,11 @@ def get_queryset(self):
                 val = get[fieldname]
                 if isinstance(Gloss._meta.get_field(fieldname), NullBooleanField):
                     val = {'0': '', '1': None, '2': True, '3': False}[val]
-                if val != '':
-                    kwargs = {key: val}
-                    qs = qs.filter(**kwargs)
```

Comment:
I'm Approving, but if you have time would you mind a comment explaining how and why this works? :)

Reply:
@jonholdsworth , when we search for a field that is in the `fieldname` list, it goes inside this code segment (for this scenario , think we are filtering for `number_incorporated`). It goes through the code starting from line 228, and since it has a value for the `val`, it enters the code segment in line 236, where you would filter the queryset based on the `**kwargs`. In this case, `kwargs` would be `number_incorporated=on`. Since `number_incorporated` is a boolean field, this would error out.

Let me know if you need more clarification. :smile:

Reply:
That's great thank you! :D